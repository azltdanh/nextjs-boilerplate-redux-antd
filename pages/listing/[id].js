import { Row, Col, Button } from 'antd';
import { getAllListingIds, getListingDetails } from '@services/listing';

export const getStaticPaths = async () => {
  const paths = await getAllListingIds();
  return {
    paths,
    fallback: false,
  };
};

export const getStaticProps = async ({ params }) => {
  const listingItem = await getListingDetails(params.id);
  return {
    props: {
      listingItem: listingItem,
    },
  };
};

const ListingItem = ({ listingItem }) => {
  const layoutWrap = {
    xs: 24,
    md: 12,
  };

  const itemWrap = {
    span: 12,
  };

  return (
    <article>
      <img src={listingItem.images[0]} alt={listingItem.title} width={360} />
      <h1>{listingItem.title}</h1>
      <Row>
        <Button type="primary" htmlType="submit" block>
          ĐẶT XE
        </Button>
      </Row>
      <Row>
        <Col {...layoutWrap}>ĐẶC ĐIỂM</Col>
        <Col {...layoutWrap}>
          <Row>
            <Col {...itemWrap}>Số ghế: {listingItem.seats}</Col>
            <Col {...itemWrap}>Truyền động: {listingItem.type}</Col>
            <Col {...itemWrap}>Nhiên liệu: {listingItem.fuel}</Col>
          </Row>
        </Col>
      </Row>
      <Row>
        <Col {...layoutWrap}>MÔ TẢ</Col>
        <Col {...layoutWrap}>{listingItem.desc}</Col>
      </Row>
    </article>
  );
};

export default ListingItem;
