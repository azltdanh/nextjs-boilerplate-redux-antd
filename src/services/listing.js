import api from './api';

export const getAllListingIds = async () => {
  const resp = await api.get('/listing');
  return resp.data.map((item) => ({
    params: {
      id: item.id.toString(),
    },
  }));
};

export const getListing = async ({ page = 1, sort = 'id', order = 'asc' }) => {
  const resp = await api.get(
    `/listing?_page=${page}&_limit=10&_sort=${sort}&_order=${order}`
  );
  return resp.data;
};

export const getListingDetails = async (id) => {
  const resp = await api.get(`/listing/${id}`);
  return resp.data;
};
