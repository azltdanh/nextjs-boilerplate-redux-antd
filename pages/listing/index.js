import Link from 'next/link';
import { getListing } from '@services/listing';

export const getServerSideProps = async () => {
  const allListing = await getListing({});
  return {
    props: {
      allListing: allListing,
    },
  };
};

const Listing = ({ allListing }) => {
  return (
    <section>
      <h2>Listing</h2>
      <ul>
        {allListing &&
          allListing.map(({ id, title }) => (
            <li key={id}>
              <Link href={`/listing/${id}`}>
                <a>{title}</a>
              </Link>
            </li>
          ))}
      </ul>
    </section>
  );
};

export default Listing;
