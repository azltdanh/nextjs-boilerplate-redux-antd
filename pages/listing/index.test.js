import * as listingServices from '@services/listing';
import mockListingItem from '@mock/listingItem';
import Listing, { getServerSideProps } from '.';

describe(Listing.name, () => {
  it('getServerSideProps', async () => {
    const getListingSpy = jest
      .spyOn(listingServices, 'getListing')
      .mockImplementation(() => Promise.resolve([mockListingItem]));
    const staticProps = await getServerSideProps();
    expect(getListingSpy).toHaveBeenCalledTimes(1);
    expect(staticProps.props.allListing).toEqual([mockListingItem]);
  });

  it('renders', () => {
    const props = {
      allListing: [mockListingItem],
    };
    const wrapper = shallow(<Listing {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
