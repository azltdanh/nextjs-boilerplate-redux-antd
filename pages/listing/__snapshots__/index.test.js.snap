// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing renders 1`] = `
<section>
  <h2>
    Listing
  </h2>
  <ul>
    <li
      key="1"
    >
      <Link
        href="/listing/1"
      >
        <a>
          KIA MORNING SI 2017
        </a>
      </Link>
    </li>
  </ul>
</section>
`;
